### PR TITLE
Make render_in_view_context forward its args to the block

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

Our `render_in_view_context` test helper uses `instance_exec` to evaluate the given block in the view context. Since `instance_exec` changes the value of `self`, instance variables, methods, etc defined in the caller's context are not available inside the block, which can be pretty confusing. For example, consider this test:

```ruby
class MyTest < Minitest::Test
  def setup
    @user = create(:user)
  end

  def test_my_component
    render_in_view_context do
      render(MyComponent.new(@user))
    end
  end
end
```

As a direct result of using `instance_exec`, `@user` will be `nil` inside the block.

### What approach did you choose and why?

Presumably to provide an escape hatch for this problem, `instance_exec` forwards its arguments when calling the block. This PR modifies `render_in_view_context` to forward _its_ arguments to `instance_exec`, so this is now possible:

```ruby
class MyTest < Minitest::Test
  def setup
    @user = create(:user)
  end

  def test_my_component
    render_in_view_context(@user) do |user|
      render(MyComponent.new(user))
    end
  end
end
```

### CODE REVIEW

File: .standard.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+ruby_version: 2.5
```
Comment (Spone) [inline .standard.yml:1]: This change seems unrelated?

**Review by joelhawksley — COMMENTED** [review]
Makes sense to me. Mind adding a regression test?

File: .standard.yml (hunk 1 above)
Comment (camertron) [reply .standard.yml:1]: Good call-out, but it actually is related. Standard wanted me to use argument forwarding, eg `render_in_view_context(...)` instead of `render_in_view_context(*args, &block)`, but arg forwarding wasn't introduced until Ruby 2.7. Since view_component supports Ruby versions all the way back to 2.5, I needed to tell standard to only apply rules that make sense for Ruby versions <= 2.5.

Comment (Spone) [reply .standard.yml:1]: Makes sense, thanks for clarifying!

File: test/sandbox/test/rendering_test.rb [hunk 2]
```diff
@@ -15,6 +15,17 @@ def test_render_in_view_context
     assert_selector("div", text: "hello,world!")
   end
 
+  def test_render_in_view_context_forwards_arguments
+    @foo = "foo"
+    @bar = "bar"
+
+    render_in_view_context(@foo, bar: @bar) do |foo, bar:|
+      render(MyComponent.new) { foo + bar }
```
Comment (joelhawksley) [inline test/sandbox/test/rendering_test.rb:23]: Not sure if it matters, but this test does not match the PR description, which had a reference to the ivar inside the `render` call 🤔

**Review by BlakeWilliams — APPROVED** [review]
Nice, I really like this solution. 👍

File: test/sandbox/test/rendering_test.rb (hunk 2 above)
Comment (camertron) [reply test/sandbox/test/rendering_test.rb:23]: The example in the PR description that uses an ivar inside `render_in_view_context` is the one that doesn't work. There's nothing we can do about ivar access (since the value of `self` is changed by `instance_exec`), so there's no way to make the example work as written. The idea behind this PR is to be able to pass data _stored_ in ivars (and elsewhere) into the block as arguments (locals). It's an imperfect workaround, but I don't think there's much else we can do.
